homeassistant/components/minecraft_server/sensor.py

```diff
@@ -0,0 +1,262 @@
+"""The Minecraft Server sensor platform."""
+
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import MinecraftServer, MinecraftServerEntity
+from .const import (
+    DOMAIN,
+    ICON_DESCRIPTION,
+    ICON_LATENCY_TIME,
+    ICON_PLAYERS_LIST,
+    ICON_PLAYERS_MAX,
+    ICON_PLAYERS_ONLINE,
+    ICON_PROTOCOL_VERSION,
+    ICON_VERSION,
+    NAME_DESCRIPTION,
+    NAME_LATENCY_TIME,
+    NAME_PLAYERS_LIST,
+    NAME_PLAYERS_MAX,
+    NAME_PLAYERS_ONLINE,
+    NAME_PROTOCOL_VERSION,
+    NAME_VERSION,
+    UNIT_DESCRIPTION,
+    UNIT_LATENCY_TIME,
+    UNIT_PLAYERS_LIST,
+    UNIT_PLAYERS_MAX,
+    UNIT_PLAYERS_ONLINE,
+    UNIT_PROTOCOL_VERSION,
+    UNIT_VERSION,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Minecraft Server sensor platform."""
+    server = hass.data[DOMAIN][config_entry.unique_id]
+
+    # Create entities list.
+    entities = [
+        MinecraftServerDescriptionSensor(hass, server),
+        MinecraftServerVersionSensor(hass, server),
+        MinecraftServerProtocolVersionSensor(hass, server),
+        MinecraftServerLatencyTimeSensor(hass, server),
+        MinecraftServerPlayersOnlineSensor(hass, server),
+        MinecraftServerPlayersMaxSensor(hass, server),
+        MinecraftServerPlayersListSensor(hass, server),
+    ]
+
+    # Add sensor entities.
+    async_add_entities(entities, True)
+
+
+class MinecraftServerSensorEntity(MinecraftServerEntity):
+    """Representation of a Minecraft Server sensor base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        server: MinecraftServer,
+        type_name: str,
+        icon: str = None,
+        unit: str = None,
+        device_class: str = None,
+    ) -> None:
+        """Initialize sensor base entity."""
+        super().__init__(hass, server, type_name, icon, device_class)
+        self._state = None
+        self._unit = unit
+
+    @property
+    def state(self) -> Any:
+        """Return sensor state."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return sensor measurement unit."""
+        return self._unit
+
+
+class MinecraftServerDescriptionSensor(MinecraftServerSensorEntity):
```

Comment:
Not a review : actually wondering if this sensor is usefull ...

Reply:
I was also asking myself the same question :) But then I thought that it's maybe useful for notifications. For example something like "Someone just joined the server <description>". This could probably also be done via templates and state attributes, but working directly with sensors is for me easier...